### PR TITLE
Add unit test for cgroup v1 v2 conversion

### PR DESCRIPTION
**Pull request by Jeffwan**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
No 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NO
```

### CODE REVIEW

**Comment by vinaykul** [comment]
@Jeffwan Thanks a lot for working on this change. 
Could you please rebase this fix and preferably add the fix for https://github.com/kubernetes/kubernetes/issues/112264 so that I can try out the integrated change before I merge it with the mothership PR https://github.com/kubernetes/kubernetes/pull/102884?

**Comment by Jeffwan** [comment]
@vinaykul Yeah. Let me make it today.

**Comment by Jeffwan** [comment]
@vinaykul I rebase the change and resolve the conflict. For the #112264, it's a lilttle bit different from what I've seen. I takes more time to fix the issue. I will update the status later.

**Comment by vinaykul** [comment]
The tests are failing for me. Does it work for you?

```
root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs# git status .
On branch restart-free-pod-vertical-scaling
Your branch is up to date with 'origin/restart-free-pod-vertical-scaling'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   pkg/kubelet/cm/cgroup_manager_linux.go
	modified:   pkg/kubelet/cm/cgroup_manager_linux_test.go

root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs# make test WHAT="./pkg/kubelet/cm"
+++ [1009 13:27:26] Running tests without code coverage and with -race
--- FAIL: TestCpuSharesToCpuWeight (0.00s)
    cgroup_manager_linux_test.go:194: Unexpected result, input: 0, expected: 0, actual: 70369281052672
--- FAIL: TestCpuWeightToCpuShares (0.00s)
    cgroup_manager_linux_test.go:220: Unexpected result, input: 0, expected: 0, actual: 1844858893260256
E1009 13:27:30.965739  383354 container_manager_linux.go:928] "Unable to get rootfs data from cAdvisor interface" err="Unable to get rootfs data from cAdvisor interface"
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/cm	0.160s
FAIL
make: *** [Makefile:184: test] Error 1
root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs# make kubelet
+++ [1009 13:27:39] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubelet (non-static)
root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs# 
root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs# make test WHAT="./pkg/kubelet/cm"
+++ [1009 13:28:09] Running tests without code coverage and with -race
--- FAIL: TestCpuSharesToCpuWeight (0.00s)
    cgroup_manager_linux_test.go:194: Unexpected result, input: 0, expected: 0, actual: 70369281052672
--- FAIL: TestCpuWeightToCpuShares (0.00s)
    cgroup_manager_linux_test.go:220: Unexpected result, input: 0, expected: 0, actual: 1844858893260256
E1009 13:28:14.051539  384107 container_manager_linux.go:928] "Unable to get rootfs data from cAdvisor interface" err="Unable to get rootfs data from cAdvisor interface"
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/cm	0.154s
FAIL
make: *** [Makefile:184: test] Error 1
root@fw0000359:~/go/src/k8s.io/kubernetes-rfpvs#
```

File: pkg/kubelet/cm/cgroup_manager_linux_test.go [hunk 1]
```diff
@@ -169,3 +169,55 @@ func TestParseSystemdToCgroupName(t *testing.T) {
...
 	}
 }
+
+func TestCpuSharesToCpuWeight(t *testing.T) {
+	testCases := []struct {
+		input    uint64
+		expected uint64
+	}{
+		{
+			input:    0,
```
Comment (vinaykul) [inline pkg/kubelet/cm/cgroup_manager_linux_test.go:179]: This is now a valid test case. Cpu shares is set to a minimum value of 2 at this time.
https://github.com/kubernetes/kubernetes/pull/102884#discussion_r662552642

File: pkg/kubelet/cm/cgroup_manager_linux_test.go [hunk 2]
```diff
@@ -169,3 +169,55 @@ func TestParseSystemdToCgroupName(t *testing.T) {
...
+	}
+}
+
+func TestCpuWeightToCpuShares(t *testing.T) {
+	testCases := []struct {
+		input    uint64
+		expected uint64
+	}{
+		{
+			input:    0,
```
Comment (vinaykul) [inline pkg/kubelet/cm/cgroup_manager_linux_test.go:205]: This is invalid test case.

**Comment by vinaykul** [comment]
@Jeffwan Unit tests and review fixes are part of the dev work unit, so it is easier for me to take care of it. I'll address the issues Mrunal has raised as it is better use of both our our time.

Please focus on fixing https://github.com/kubernetes/kubernetes/issues/112264
Thanks.

**Pull request closed without merging**